cmd/estimate/reviewTime.go

```diff
@@ -0,0 +1,271 @@
+		}
+
+		result += float64(file.GetAdditions()) * estimate * config.Base
+		result += float64(file.GetDeletions()) * estimate * config.Deletion
+	}
+	return int(result)
+}
+
+func parseConfig(configPath string, cf *configFile) error {
+	yamlFile, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("error reading file at '%s': %v", configPath, err)
+	}
+	err = yaml.Unmarshal(yamlFile, cf)
+	if err != nil {
+		return err
+	}
+	if addLabel && len(cf.Labels) == 0 {
+		return emptyLabelsError
+
+	}
+	if len(cf.Extensions) == 0 {
+		klog.Warningf("'extensions' list not specified")
+	}
+	if err != nil {
+		return fmt.Errorf("error during unmarshaling %v", err)
+	}
+	return nil
+}
+
+func addLabelToPR(client *github.Client, reviewTime int) error {
+	existingLabels, err := listLabels(client)
+	if err != nil {
+		return err
+	}
+	calculatedLabel, err := getLabelBasedOnTime(reviewTime)
+	if err != nil {
+		return err
+	}
+	klog.Infof("Calculated label '%s'\n", calculatedLabel.Name)
+
+	for _, existingLabel := range existingLabels {
+		if *existingLabel.Name == calculatedLabel.Name {
+			klog.Infof("The issue already has the same label '%s'. Skipping addition.\n", *existingLabel.Name)
+			return nil
+		}
+		// Remove outdated label if the estimation changed
+		for _, timeLabel := range config.Labels {
+			if timeLabel.Name == *existingLabel.Name {
+				_, err := client.Issues.RemoveLabelForIssue(context.Background(), owner, repository, prNumber, timeLabel.Name)
+				if err != nil {
+					return err
+				}
+				klog.Infof("Removed outdated label '%s'", timeLabel.Name)
+			}
+		}
+	}
+	_, _, err = client.Issues.AddLabelsToIssue(context.Background(), owner, repository, prNumber, []string{calculatedLabel.Name})
+	if err != nil {
+		return err
+	}
+	klog.Infof("Added label '%s'\n", calculatedLabel.Name)
+	return nil
+}
+
+func listLabels(client *github.Client) ([]*github.Label, error) {
+	labels, _, err := client.Issues.ListLabelsByIssue(context.Background(), owner, repository, prNumber, nil)
+	if err != nil {
+		return nil, err
+	}
+	return labels, nil
+}
+
+func getLabelBasedOnTime(reviewTime int) (*TimeLabel, error) {
+	if len(config.Labels) == 0 {
+		return nil, emptyLabelsError
+	}
+	maxLabel := TimeLabel{}
+	for _, label := range config.Labels {
+		if label.Time < reviewTime && label.Time > maxLabel.Time {
+			maxLabel = label
+		}
+	}
+	return &maxLabel, nil
+}
+
+func init() {
+	EstimateTimeToReviewCmd.Flags().StringVar(&owner, "owner", "redhat-appstudio", "owner of the repository")
+	EstimateTimeToReviewCmd.Flags().StringVar(&repository, "repository", "e2e-tests", "name of the repository")
+	EstimateTimeToReviewCmd.Flags().IntVar(&prNumber, "number", 1, "number of the pull request")
+	err := EstimateTimeToReviewCmd.MarkFlagRequired("number")
+	if err != nil { // silence golangci-lint
+		return
+	}
+	EstimateTimeToReviewCmd.Flags().StringVar(&configPath, "config", "", "path to the yaml config file")
```

Comment:
can we leverage viper's functionality of config parsing (and providing a default path to the config)? example [here](https://github.com/redhat-appstudio/qe-tools/blob/b4ac4d4fac508f5e59926ff03dc06cccb1367e5f/cmd/prowjob/healthCheck.go#L66-L74) 
then we don't need to mark the config flag as required

Reply:
Cool, didn't notice that in the viper quick start doc... Thanks, I'll take a look, seems like a great idea